backend/src/main/java/me/magicbudget/controller/GroupController.java

```
@@ -0,0 +1,109 @@
+package me.magicbudget.controller;
+
+
+import me.magicbudget.dto.incomingrequest.AddUserToGroupRequest;
+import me.magicbudget.dto.incomingrequest.TransactionRequest;
+import me.magicbudget.dto.outgoingresponse.GroupResponse;
+import me.magicbudget.dto.outgoingresponse.SplitTransactionResponse;
+import me.magicbudget.service.GroupService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RestController
+@RequestMapping("api/{userid}/group")
+public class GroupController {
+
+  private final GroupService groupService;
+
+  public GroupController(GroupService groupService) {
+    this.groupService = groupService;
+  }
+
+  @PostMapping("/create-group")
+  public ResponseEntity<String> createGroup(@PathVariable("userid") String userId,
+      @RequestBody String groupName) {
+    try {
+      groupService.createGroup(groupName, userId);
+      return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+    catch (Exception e) {
```

Again suppressing errors